include/IOSOperator.h

```diff
@@ -26,12 +26,23 @@ class IOSOperator : public virtual IReferenceCounted
 	}
 
 	//! Copies text to the clipboard
+	//! \param text: text in utf-8
 	virtual void copyToClipboard(const c8* text) const = 0;
 
+	//! Copies text to the primary selection
+	//! This is a no-op on some platforms.
+	//! \param text: text in utf-8
+	virtual void copyToPrimarySelection(const c8* text) const = 0;

```

Comment:
Breaks binary compatibility, is that OK?

Reply:
Specifically, Minetest compiled against current HEAD would crash if ran against this version, because of VMT changes. If that is problematic, move the new virtual methods to the bottom like you would do for new fields.

Reply:
TBH, I know almost nothing about binary compatibility in C++. Is it *guaranteed* that code that expect old vtables works with newer vtables if the new virtual member function is added at the bottom? If you know some references where such things are written down, I'd be very happy to read from them.

Also, I don't know if a binary compatibility break is ok. @sfan5 ?
And I haven't looked into how minetest uses its irrlicht revision versions.

Reply:
We don't care about binary compatibility. (each release is a higher [revision](https://github.com/minetest/irrlicht/commit/a0af653c3d1d0e4a17c9c9408ff7eb0fc697ad4b) and bumps the SOVERSION)
We'll stop caring about source compatibility soon too when the submodule stuff gets in.